source/slang/syntax.cpp

```diff
@@ -1205,9 +1205,7 @@ void Type::accept(IValVisitor* visitor, void* extra)
         if (!this) return nullptr;
 
         int diff = 0;
-        RefPtr<Substitutions> outerSubst;
-        if (outer)
-            outerSubst = outer->SubstituteImpl(subst, &diff);
+        RefPtr<Substitutions> outerSubst = outer ? outer->SubstituteImpl(subst, &diff) : nullptr;
```

Comment:
Why don’t these recursive call sites use `substituteSubstitutions()` as well? You mention in the commit message that it is only supposed to be applied to the “head” of a substitution list, but I don’t think I understand why.